pkg/rbac/middleware.go

```diff
@@ -5,19 +5,43 @@ import (
 	"encoding/json"
 	"net/http"
 	"strings"
+	"sync"
+	"time"
 
 	"github.com/stolostron/search-v2-api/pkg/config"
 	authv1 "k8s.io/api/authentication/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog/v2"
 )
 
+type RbacCache struct {
+	UserRbac []*UserRbac
```

Comment:
Suggestion: Maybe `Users []*UserRbac` is more intuitive and avoids confusion with the UserRbac type.

Reply:
Done.